hadoop-hdds/docs/content/design/ozone-sts.md

```diff
@@ -0,0 +1,147 @@
+            "Action": "s3:GetObject",
+            "Resource": "arn:aws:s3:::example-bucket/*"
+        }
+    ]
+}
+
+```
+
+## 3.3 SessionToken Format
+
+As mentioned above, one of the return values from the AssumeRole call will be the sessionToken. To support not
+storing temporary credentials server-side in Ozone, the sessionToken will comprise various components needed to validate
+subsequent S3 calls that use the token.  The sessionToken will have the following information encoded:
+
+- The originalAccessKeyId - this is the Kerberos identity of the user that created the sessionToken via the AssumeRole call.
+When the temporary credentials are used to make S3 API calls, this Kerberos identity (in conjunction with the role permissions and
+optional session policy) will be used to authorize the call.
+- The roleArn - the role used in the original AssumeRole call
+- The encrypted secretAccessKey - this will be used to validate the AWS signature when the temporary credentials are used 
+to make S3 API calls
+- (Optional) sessionPolicy - when using the RangerOzoneAuthorizer, if Ranger successfully authorizes the AssumeRole call,
+it will return a String representing the resources (i.e. buckets, keys, etc.) and permissions (i.e. ACLType) that the token
+has been granted access to.  This sessionPolicy will be included in the sessionToken sent back to the client so it can be sent to Ranger to 
+authorize subsequent S3 API calls that use the sessionToken.
+- HMAC-SHA256 signature - used to ensure the sessionToken was created by Ozone and was not altered since it was created.
+- The expiration time of the token (via `ShortLivedTokenIdentifier#getExpiry()`)
+- The UUID of the secret key used to sign the sessionToken and encrypt the secretAccessKey (via `ShortLivedTokenIdentifier#getSecretKeyId()`)
+
+## 3.5 STS Token Revocation
+
+In the rare event temporary credentials need to be revoked (ex. for security reasons), a table in RocksDB will be created
+to store revoked tokens, and a command-line utility will be created to add tokens to the table.  A background cleaner service
+will be created to run every 3 hours to delete revoked tokens that have been in the table for more than 12 hours.
+
+## 3.6 Overall Flow
+
+The following section outlines the overall flow when using STS in Ozone:
+
+- An authorized user (having permanent S3 credentials) makes the AssumeRole STS call to Ozone.
+- If successful, Ozone responds with the temporary credentials.
+- A client makes S3 API calls with the temporary credentials for up to as long as the credentials last.  
+- When Ozone receives an S3 api call using temporary credentials, it will perform the following checks:
+  - Ensure the token is not expired
+  - Ensure the token is not revoked
+  - Validate the HMAC-SHA256 signature in the sessionToken
+  - Decrypt the secretAccessKey from the sessionToken and validate the AWS signature
+  - Authorize the call with either RangerOzoneAuthorizer or OzoneNativeAuthorizer
```

Comment:
We should document how each authorizer will handle the tokens. Especially any work required from the Ranger side.